File: Saving_Account.java

import java.util.Scanner;;
public class Saving_Account 
{
    double P; 
    double R;
    double Intrest;
    double T;
    double W=0;
    double F;
    Scanner Input = new Scanner(System.in);
    public Saving_Account(double Principle, double rate, double CI, double Time, double Wallet) // Wallet is for personal money
    {
        P = Principle;
        R = rate;
        Intrest = CI;
        Time = T;
        W=Wallet;
    }

    public Saving_Account() // For my object because laziness
    {

    }

    public static void Main(String [] args)
    {
        
    }

    public double Intrest(double Principle, double rate, double CI, double Time)//this is for the compound intrest formula.
    {   //  CI is amount of times per year your intrest compounds. 
        // And Time is the variable for how many years have passed in percentange
        double FV = Math.pow(Principle*(1 + rate/CI), Time * CI); //FV = future values in the formula P(r + n/r)
        System.out.println("" + FV);
        return F = FV;
    }

    public double Deposit(double D)
    {
        return P+=D;
    }

    public double Withdrawl(double WD)
    {
        WD=W;
        return P-=WD;
    }
}

File: src/main/java/saving_account/SavingAccount.kt
package saving_account

import kotlin.math.pow

// Wallet is for personal money
class SavingAccount(var principle: Double, var rate: Double, var compoundInterest: Double, var time: Double) {
    var wallet = 0.0
    var f = 0.0

    //This is for the compound interest formula.
    fun interest(principle: Double, rate: Double, compoundInterest: Double, time: Double): Double {
        // time is the variable for how many years have passed in percentage
        val futureValue = (principle * (1 + rate / compoundInterest)).pow(time * compoundInterest)
        // future values in the formula P(r + n/r)
        println(futureValue)
        return futureValue.also { f = it }
    }

    fun deposit(D: Double): Double {
        return D.let { principle += it; principle }
    }

    fun withdrawal(WD: Double): Double {
        return WD.coerceAtMost(wallet).let { principle -= it; principle }
    }
}